src/Expander.hs

```diff
@@ -1191,17 +1192,15 @@ runTask (tid, localData, task) = withLocal localData $ do
         Just _v -> do
           bind b $ EVarMacro x
           forkExpandSyntax (ExprDest t dest) stx
-    ExpandDecl dest sc stx ph ->
-      expandOneDeclaration sc dest stx ph
-    ExpandMoreDecls dest sc stx waitingOn -> do
-      readyYet <- view (expanderDeclPhases . at waitingOn) <$> getState
+    ExpandDeclTree dest sc stx vp ->
+      expandOneDeclTreeNode sc dest stx vp
+    ExpandDependentDeclTree dest stx waitingOn -> do
+      readyYet <- view (expanderDeclValidities . at waitingOn) <$> getState
       case readyYet of
         Nothing ->
           stillStuck tid task
-        Just (SpecificPhase p) ->
-          forkExpandDecls dest (addScope p stx sc)
-        Just AllPhases ->
-          forkExpandDecls dest (addScope' stx sc)
+        Just scopeSet ->
+          forkExpandDecls dest (addScopes stx scopeSet)
```

Comment:
We're now adding all the scopes in the `ScopeSet` to `stx`, not just `sc`. This task is no longer receiving that `sc`, because we now receive the scope from the `DeclValidityPtr`.